screenpipe-vision/src/core.rs

```diff
@@ -156,47 +162,39 @@ pub async fn continuous_capture(
         previous_image = Some(Arc::new(image.clone()));
         // debug!("ocr_task_running {} BEFORE if if !ocr_task_running.load(Ordering::SeqCst)", ocr_task_running.load(Ordering::SeqCst));
 
-        if !ocr_task_running.load(Ordering::SeqCst) {
-            // debug!("max_avg_frame {} before if let Some(", max_avg_value);
-            if let Some(max_avg_frame) = max_average.take() {
-                // Use take() to move out the value
-                let ocr_task_data = OcrTaskData {
-                    image: max_avg_frame.image.clone(),
-                    frame_number: max_avg_frame.frame_number,
-                    timestamp: max_avg_frame.timestamp,
-                    result_tx: result_tx.clone(),
-                };
-
-                let previous_text_json_clone = previous_text_json.clone();
-                let ocr_task_running_clone = ocr_task_running.clone();
+        // debug!("max_avg_frame {} before if let Some(", max_avg_value);
+        if let Some(max_avg_frame) = max_average.take() {
+            // Use take() to move out the value
+            let ocr_task_data = OcrTaskData {
+                image: max_avg_frame.image.clone(),
+                frame_number: max_avg_frame.frame_number,
+                timestamp: max_avg_frame.timestamp,
+                result_tx: result_tx.clone(),
+            };
 
-                ocr_task_running.store(true, Ordering::SeqCst);
-                // debug!("ocr_task_running {}", ocr_task_running.load(Ordering::SeqCst));
-                let ocr_engine_clone = ocr_engine.clone();
-                tokio::spawn(async move {
```

Comment:
also this thread spawn was not necessary at all?

Reply:
no effect because we await inside 

could have been useful if we didnt await the inside func maybe? i do that sometime in js but i think async here is slightly different